### PR TITLE
Cart

### PR DESCRIPTION
**Pull request by Developers-codz**
## Files to check
`src/component/navbar/navbar.jsx`
`src/component/product/Productlisting.jsx`
`src/context/cart-context.js`
`src/pages/cart/Cart.jsx`
`src/reducer/cart-reducer.js`

Thank you for reviewing PR :)

### CODE REVIEW

File: src/context/cart-context.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+          product: item,
+        },
+        {
+          headers: {
+            authorization: encodedToken,
+          },
+        }
+      );
+      if (response.status === 201) {
+        console.log(response.data.cart);
```
Comment (vrinda-mahajan) [inline src/context/cart-context.js:28]: remove` console.log` if not required.

File: src/pages/cart/Cart.jsx [hunk 2]
```diff
@@ -1,8 +1,107 @@
...
+        {cartData.map(
+          ({ _id, image, name, price, qty, rating, description }) => {
+            return (
+              <div className="cart-items" key={_id}>
+                <img src={image} alt={name} />
+
+                <div className="cart-product-detail">
+                  <div
+                    className="del-btn-cart"
+                    onClick={(e) => removeFromCartHandler(_id, qty)}
```
Comment (vrinda-mahajan) [inline src/pages/cart/Cart.jsx]: You can remove `e` from onClick.

File: src/pages/cart/Cart.jsx [hunk 3]
```diff
@@ -1,8 +1,94 @@
...
+import { shoppingImage } from "assets/svgs";
+import { useAuth } from "context/auth-context";
 const Cart = () => {
   useDocumentTitle("Cart");
-  return <>Cart</>;
+  const {
+    authState: { cart },
+  } = useAuth();
+  return cart.length === 0 ? (
+    <div style={{ textAlign: "center" }}>
```
Comment (arjun-computer-geek) [inline src/pages/cart/Cart.jsx:16]: Inline css is not good i think

**Review by arjun-computer-geek — APPROVED** [review]
Code is cleaner. All code are readable and nicely formatted

File: src/reducer/cart-reducer.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+export const cartReducer = (state, action) => {
+  switch (action.type) {
+    case "set_to_cart":
+      return { ...state, cartData: action.payload };
+    case "set_qtty":
+      return {
+        ...state,
+        cartData: action.payload,
```
Comment (vrinda-mahajan) [inline src/reducer/cart-reducer.js]: As both cases are doing the same thing, You can combine them by giving a common type to both cases like `CHANGE_CART`. Also, add the default block to the switch statement.

File: src/pages/cart/Cart.jsx (hunk 3 above)
Comment (Developers-codz) [reply src/pages/cart/Cart.jsx:16]: Will look into that, When I do code refactor Thank you :)

File: src/pages/cart/Cart.jsx [hunk 5]
```diff
@@ -1,8 +1,107 @@
...
-  return <>Cart</>;
+  const {
+    cartState: { cartData },
+    removeFromCartHandler,
+    cartCount,
+    productQtyIncreaseHandler,
+    productQtyDecreaseHandler,
+  } = useCart();
+  return cartData.length === 0 ? (
+    <div style={{ textAlign: "center" }}>
```
Comment (vrinda-mahajan) [inline src/pages/cart/Cart.jsx:16]: Try to avoid inline styling.

File: src/component/product/Productlisting.jsx [hunk 6]
```diff
@@ -1,10 +1,29 @@
...
+    cartState: { cartData },
+  } = useCart();
   const { setWishCount, setWishList, wishList } = useWishlist();
   const isInWishList = (id) => wishList.find(({ _id }) => _id == id);
+  const isInCartList = (id) => cartData.find(({ _id }) => _id == id);
   var classNames = require("classnames");
+
+  const clickHandler = (e, product) => {
+    addToCartHandler(product);
+  };
```
Comment (vrinda-mahajan) [inline src/component/product/Productlisting.jsx:25]: I guess `event` is not required here.

**Review by vrinda-mahajan — APPROVED** [review]
Added a few suggestions.

File: src/context/cart-context.js (hunk 1 above)
Comment (Developers-codz) [reply src/context/cart-context.js:28]: Will remove after completing

File: src/pages/cart/Cart.jsx (hunk 2 above)
Comment (Developers-codz) [reply src/pages/cart/Cart.jsx]: Done

File: src/reducer/cart-reducer.js (hunk 4 above)
Comment (Developers-codz) [reply src/reducer/cart-reducer.js]: Done , Thanks for pointing those things out

**Pull request merged**